daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -605,7 +608,21 @@ private[archive] class DecodeV1(minor: LV.Minor) extends Decode.OfPackage[PLF.Pa
           throw ParseError("Kind.SUM_NOT_SET")
       }
 
-    private[lf] def decodeType(lfType: PLF.Type): Type =
+    private[archive] def decodeType(lfType: PLF.Type): Type =
+      if (versionIsOlderThan(LV.Features.internedTypes))
+        uncheckedDecodeType(lfType)
+      else
+        lfType.getSumCase match {
+          case PLF.Type.SumCase.INTERNED =>
+            internedTypes.applyOrElse(
+              lfType.getInterned,
+              (index: Int) => throw ParseError(s"invalid internedTypes table index $index"),
+            )
+          case otherwise =>
+            throw ParseError(s"$otherwise is not supported outside type interning table")
```

Comment:
Just to confirm I understand this correctly: In DAML-LF 1.dev every type outside of the type interning table has to be a reference into the type interning table and types inside the type interning table can be arbitrarily complex and contain references into the table at any level.

Reply:
Yes. 
* `decodeType` enforces (for 1.dev only) that `lfType` is a reference to the the type interning table. (should be use to decode type outside the interning table)
* `uncheckedDecodeType` allows `lfType` to be arbitrarily complex and contain references into the table at any level.  (should be use to decode type inside the interning table)

Reply:
Thanks for confirming.